examples/generative/corrdiff/training/training_loop.py

```diff
@@ -264,12 +283,17 @@ def training_loop(
             g["lr"] = optimizer_kwargs["lr"] * min(
                 cur_nimg / max(lr_rampup_kimg * 1000, 1e-8), 1
             )  # TODO better handling (potential bug)
+            if patch_shape_x != img_shape_x:
+                g["lr"] *= lr_decay ** ((cur_nimg - lr_rampup_kimg * 1000) // 5e6)
             wb.log({"lr": g["lr"]}, step=cur_nimg)
         for param in net.parameters():
             if param.grad is not None:
                 torch.nan_to_num(
                     param.grad, nan=0, posinf=1e5, neginf=-1e5, out=param.grad
                 )
+        grad_norm = torch.nn.utils.clip_grad_norm_(
```

Comment:
Note that this is changing the default behavior. Previously grad norm was not used but now you are using a non-zero threshold even without patching.

Reply:
I think we had this conversion during the previous round of review: whether or not we should support gradient clipping and learning rating scheduling for non patch-based training. By default, the code does not use. Shall we set a sufficiently large value for gradient clipping threshold in the config file when is not used in order to keep it consistent?

Reply:
I'm okay with some backwards breaking changes to the training.

Reply:
Never mind. I have set the default config to not using graident clipping